src/main/java/com/parsing/pdf/parsing/ParsingPDFService.java

```diff
@@ -0,0 +1,142 @@
+        }
+        obj.put("text", builder);
+        return obj.toString();
+    }
+
+    private List<Row> findStrippedTextFromFile(MultipartFile file) {
+        try {
+            PDDocument document = PDDocument.load(file.getBytes());
+            PDFTextStripper stripper = new PDFTextStripper();
+            String strippedText = stripper.getText(document);
+            if (strippedText.trim().isEmpty()) {
+                return Collections.emptyList();
+            }
+        } catch (Exception e) {
+            throw new PDFParsingException(String.format("Parsing for file: %s failed.", file.getOriginalFilename()));
+        }
+        return Collections.emptyList();
+    }
+
+    private List<Row> extractTextFromScannedDocument(String fileTableName) throws TesseractException {
+        ITesseract itesseract = new Tesseract();
+        itesseract.setDatapath(getPathTessData());
+        itesseract.setLanguage("ukr+eng");
+        final Mat tableMat = Imgcodecs.imread(fileTableName);
+        for (Row row : table) {
+            for (Column column : row.getColumns()) {
+                Mat mat = fillSquareWhitePixel(tableMat, column.getRectangle());
+                Imgcodecs.imwrite(table.indexOf(row) + " " + row.getColumns().indexOf(column) + ".png", mat);
+            }
+        }
+        for (Row row : table) {
+            for (Column column : row.getColumns()) {
+                String filename = table.indexOf(row) + " " + row.getColumns().indexOf(column) + ".png";
+                String result = itesseract.doOCR(new File(filename));
+                column.setParsingResult(result);
+                log.info(filename + " = " + result);
+            }
+        }
+        return table;
+    }
+
+    private Mat fillSquareWhitePixel(Mat image, Rectangle rectangle) {
+        Mat result = image.clone();
+        double[] whitePixel = {WHITE_PIXEL_CODE, WHITE_PIXEL_CODE, WHITE_PIXEL_CODE};
+        double xLeftUp = rectangle.getX() + OFFSET;
+        double yLeftUp = rectangle.getY() + OFFSET;
+        double xRightDown = xLeftUp + rectangle.getWidth() - OFFSET;
+        double yRightDown = yLeftUp + rectangle.getHeight() - OFFSET;
+        for (int row = 0; row < image.rows(); row++) {
+            for (int column = 0; column < image.cols(); column++) {
+                if (xLeftUp < column && column < xRightDown && yLeftUp < row && row < yRightDown) {
+                } else {
+                    result.put(row, column, whitePixel);
+                }
+            }
+        }
+        return result;
+    }
+
+    private String getPathTessData() {
+        Path currentPathPosition = Paths.get("").toAbsolutePath();
+        File pdfDir = new File(currentPathPosition + DIR_TO_READ_TESSDATA);
+        if (!pdfDir.exists()) {
+            pdfDir.mkdir();
+        }
+        return currentPathPosition.toAbsolutePath() + DIR_TO_READ_TESSDATA;
+    }
+
+    private String getLastPagePDF(PDDocument document) throws IOException {
```

Comment:
should we take 2 pages?

Reply:
At the moment I created task for this problem - NEP-28